CMakeLists.txt

```diff
@@ -279,34 +289,38 @@ set(ALLOCATOR "" CACHE STRING
   "specify memory allocator to use. currently tcmalloc, tcmalloc_minimal, \
 jemalloc, and libc is supported. if not specified, will try to find tcmalloc, \
 and then jemalloc. If neither of then is found. use the one in libc.")
-if(ALLOCATOR)
-  if(${ALLOCATOR} MATCHES "tcmalloc(_minimal)?")
-    find_package(gperftools REQUIRED)
-    set(HAVE_LIBTCMALLOC ON)
-  elseif(${ALLOCATOR} STREQUAL "jemalloc")
-    find_package(JeMalloc REQUIRED)
-    set(HAVE_JEMALLOC 1)
-  elseif(NOT ALLOCATOR STREQUAL "libc")
-    message(FATAL_ERROR "Unsupported allocator selected: ${ALLOCATOR}")
-  endif()
-else(ALLOCATOR)
-  find_package(gperftools)
-  set(HAVE_LIBTCMALLOC ${gperftools_FOUND})
-  if(NOT gperftools_FOUND)
-    find_package(JeMalloc)
-  endif()
-  if(gperftools_FOUND)
-    set(ALLOCATOR tcmalloc)
-  elseif(JeMalloc_FOUND)
-    set(ALLOCATOR jemalloc)
-  else()
-    if(NOT FREEBSD)
-      # FreeBSD already has jemalloc as its default allocator
-      message(WARNING "tcmalloc and jemalloc not found, falling back to libc")
+if(NOT WIN32)
+  if(ALLOCATOR)
+    if(${ALLOCATOR} MATCHES "tcmalloc(_minimal)?")
+      find_package(gperftools REQUIRED)
+      set(HAVE_LIBTCMALLOC ON)
+    elseif(${ALLOCATOR} STREQUAL "jemalloc")
+      find_package(JeMalloc REQUIRED)
+      set(HAVE_JEMALLOC 1)
+    elseif(NOT ALLOCATOR STREQUAL "libc")
+      message(FATAL_ERROR "Unsupported allocator selected: ${ALLOCATOR}")
     endif()
-    set(ALLOCATOR "libc")
-  endif(gperftools_FOUND)
-endif(ALLOCATOR)
+  else(ALLOCATOR)
+    find_package(gperftools)
+    set(HAVE_LIBTCMALLOC ${gperftools_FOUND})
+    if(NOT gperftools_FOUND)
+      find_package(JeMalloc)
+    endif()
+    if(gperftools_FOUND)
+      set(ALLOCATOR tcmalloc)
+    elseif(JeMalloc_FOUND)
+      set(ALLOCATOR jemalloc)
+    else()
+      if(NOT FREEBSD)
+        # FreeBSD already has jemalloc as its default allocator
+        message(WARNING "tcmalloc and jemalloc not found, falling back to libc")
+      endif()
+      set(ALLOCATOR "libc")
+    endif(gperftools_FOUND)
+  endif(ALLOCATOR)
+else()
```

Comment:
Why would this not fit in the original testing?
If I run thru the conditionals I think I would endup with `libc`?

Reply:
We're cross-compiling yet it was retrieving the Linux binaries, gperftools I think.

Reply:
@petrutlucian94 
Could also choose the option to set ALLOCATOR=libc on the cmake CLI?
Or otherwise it in the `else(ALLOCATOR)` block

Reply:
It might do the same thing now, but wrapping it like that seemed safer/easier so that it won't be broken by mistake in the future.

Reply:
Then better put a comment with it.
Now you have disabled setting ALLOCATOR on WIN32 platorms.

Reply:
Ok, will do.